sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,163 @@
+    let settings = schnauzer::get_settings(&args.socket_path).context(error::SettingsError)?;
+
+    debug!("settings = {:#?}", settings.settings);
+    let ecs = settings
+        .settings
+        .and_then(|s| s.ecs)
+        .context(error::ModelError)?;
+
+    let mut config = ECSConfig {
+        cluster: ecs.cluster,
+        privileged_disabled: ecs.allow_privileged_containers.map(|s| !s),
+        ..Default::default()
+    };
+    if let Some(os) = settings.os {
+        config
+            .instance_attributes
+            .insert(VARIANT_ATTRIBUTE_NAME.to_string(), os.variant_id);
+        config.instance_attributes.insert(
+            VERSION_ATTRIBUTE_NAME.to_string(),
+            os.version_id.to_string(),
+        );
+    }
+    if let Some(attributes) = ecs.instance_attributes {
+        for (key, value) in attributes {
+            config
+                .instance_attributes
+                .insert(key.to_string(), value.to_string());
+        }
+    }
+
+    let serialized = serde_json::to_string(&config).unwrap();
+    debug!("serialized = {}", serialized);
+
+    let config_path = PathBuf::from(DEFAULT_ECS_CONFIG_PATH);
+    write_to_disk(config_path, serialized).context(error::FSError {
+        path: DEFAULT_ECS_CONFIG_PATH,
+    })?;
+    Ok(())
+}
+
+/// Writes the rendered data at the proper location
+fn write_to_disk<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) -> std::io::Result<()> {
+    if let Some(dirname) = path.as_ref().parent() {
+        fs::create_dir_all(dirname)?;
+    };
+
+    fs::write(path, contents)
+}
+
+// Stores user-supplied arguments.
+struct Args {
+    socket_path: String,
+}
+
+fn parse_args(args: env::Args) -> Args {
+    let mut socket_path = None;
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--socket-path" => {
+                socket_path = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --socket-path")),
+                )
+            }
+            _ => usage(),
+        }
+    }
+    Args {
+        socket_path: socket_path.unwrap_or_else(|| DEFAULT_API_SOCKET.to_string()),
+    }
+}
+
+// Prints a more specific message before exiting through usage().
+fn usage_msg<S: AsRef<str>>(msg: S) -> ! {
+    eprintln!("{}\n", msg.as_ref());
+    usage();
+}
+
+// Informs the user about proper usage of the program and exits.
+fn usage() -> ! {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {}
+            [ (-s | --socket-path) PATH ]
+
+    Socket path defaults to {}",
+        program_name, DEFAULT_API_SOCKET
+    );
+    process::exit(2);
+}
+
+type Result<T> = std::result::Result<T, SettingsApplierError>;
+
+mod error {
+    use snafu::Snafu;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub(super)")]
+    pub(super) enum SettingsApplierError {
```

Comment:
```suggestion
    pub(super) enum Error {
```
It's already scoped to this module, don't need to repeat the name :)

Reply:
Done